.github/workflows/pr_workflow.yml

```diff
@@ -25,19 +30,69 @@ jobs:
         run: |
           echo "Output: ${{ steps.check_ref.outputs.ref_contains_rc }}"
 
-  build:
+  make-or-update-pr:
     runs-on: ubuntu-latest
+    permissions: write-all
+    needs: checkRef
+    if: needs.checkRef.outputs.is_rc == '1'
 
     steps:
-      - name: Check if comment contains "Release notes"
-        if: github.ref == 'refs/heads/master'
-        uses: actions/github-script@v4
+      - name: Find Comment
+        uses: peter-evans/find-comment@v2
+        id: fc
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-includes: Release notes
+
+      - name: Create comment link
+        id: create_link
+        run: |
+          echo "COMMENT_LINK=https://api.github.com/repos/${{ github.repository }}/issues/comments/${{ steps.fc.outputs.comment-id }}" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Extract version from branch name
+        id: extract_version
+        run: |
+          VERSION=${{ github.head_ref }}
+          echo "::set-output name=version::${VERSION#*rc/}"
+
+      - uses: tibdex/github-app-token@v1
+        id: generate-token
+        with:
+          app_id: ${{ secrets.PR_APP_ID }}
+          private_key: ${{ secrets.PR_APP_TOKEN }}
+      
+      - name: Run Python script
+        run: |
+          import requests
+          import sys
+          import os
+      
+          def parse_base_params(comment_link):
+            response = requests.get(comment_link).json()
+            try:
+              lines = response["body"].split("\n")
+            except:
+              print("Please, create a comment with 'Release time: YYYY-MM-DDTHH:MM:SSZ', 'Release severity: Major | Critical | Normal' and 'Release notes:'")
+              exit(1)
+            time = next((s.split(': ')[1] for s in lines if s.startswith('Release time: ')), '').strip()
+            severity = next((s.split(': ')[1] for s in lines if s.startswith('Release severity: ')), '').strip()
+
+            if not time or not severity:
+              print(f"Current values: Time - '{time}', Severity - '{severity}'")
+              print("Time or severity cannot be empty, please set it in the PR comment with lines 'Release time:' and 'Release severity:'")
+              sys.exit(1)
+
+            with open(os.getenv('GITHUB_ENV'), 'a') as f:
+                f.write(f'TIME={time}\n')
+                f.write(f'SEVERITY={severity}\n')
+          parse_base_params(os.getenv("COMMENT_LINK"))
```

Comment:
I suggest to extract it into a separate .py file

Reply:
extracted to separate file, keep it in .github repo, checked on test rc/3.2.1